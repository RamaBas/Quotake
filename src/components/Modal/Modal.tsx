import React from 'react';
import { Modal, Box, IconButton } from '@mui/material';
import { Close } from '@mui/icons-material';

interface BaseModalProps {
  open: boolean;
  onClose: () => void;
  children: React.ReactNode;
  width?: string | number;
}

const BaseModal: React.FC<BaseModalProps> = ({ open, onClose, children, width = 400 }) => {
  return (
    <Modal
      open={open}
      onClose={onClose}
      aria-labelledby="modal-title"
      aria-describedby="modal-description"
      data-testid="modal-content"
    >
      <Box
        sx={{
          position: 'absolute',
          top: '50%',
          left: '50%',
          transform: 'translate(-50%, -50%)',
          width: width,
          bgcolor: 'background.paper',
          boxShadow: 24,
          p: 4,
          borderRadius: 2,
          outline: 'none'
        }}
      >
        <IconButton
          aria-label="close"
          onClick={onClose}
          sx={{
            position: 'absolute',
            right: 8,
            top: 8,
            color: (theme) => theme.palette.grey[500],
          }}
        >
          <Close />
        </IconButton>
        {children}
      </Box>
    </Modal>
  );
};

export default BaseModal;