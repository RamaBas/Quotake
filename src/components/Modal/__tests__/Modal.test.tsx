import { render, screen, fireEvent } from '@testing-library/react';
import Modal from '../Modal';

jest.mock('@mui/icons-material', () => ({
  Close: () => <div>CloseIcon</div>,
}));

describe('Modal Component', () => {
  const mockOnClose = jest.fn();
  const defaultProps = {
    open: true,
    onClose: mockOnClose,
    children: <div>Modal Content</div>,
  };

  it('renders correctly when open', () => {
    const { container } = render(<Modal {...defaultProps} />);
    
    const modalContent = screen.getByTestId('modal-content');
    expect(modalContent).toBeInTheDocument();
    
    expect(screen.getByText('Modal Content')).toBeInTheDocument();
    
    // Snapshot test
    expect(container.firstChild).toMatchSnapshot();
  });

  it('does not render when closed', () => {
    render(<Modal {...defaultProps} open={false} />);
    
    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
  });

  it('calls onClose when clicking the close button', () => {
    render(<Modal {...defaultProps} />);
    
    const closeButton = screen.getByLabelText('close');
    fireEvent.click(closeButton);
    
    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });
});