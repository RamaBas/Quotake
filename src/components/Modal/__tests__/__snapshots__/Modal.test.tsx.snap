// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders correctly when open 1`] = `null`;
